tools/axivion/axivion_config.json

```diff
@@ -0,0 +1,1742 @@
+        },
+        "Metric-SwitchComplexity": {
+            "_active": true
+        },
+        "Metric-TokenFileMetric.Check": {
+            "_active": true
+        },
+        "Metric-TokenFileMetric.Debug": {
+            "_active": true
+        },
+        "Metric-TokenFileMetric.Fixme": {
+            "_active": true
+        },
+        "Metric-TokenFileMetric.Hack": {
+            "_active": true
+        },
+        "Metric-TokenFileMetric.Todo": {
+            "_active": true
+        },
+        "Metric-TokenFileMetric.XXX": {
+            "_active": true
+        },
+        "Metric-TokenMetric.Check": {
+            "_active": true
+        },
+        "Metric-TokenMetric.Debug": {
+            "_active": true
+        },
+        "Metric-TokenMetric.Fixme": {
+            "_active": true
+        },
+        "Metric-TokenMetric.Hack": {
+            "_active": true
+        },
+        "Metric-TokenMetric.Todo": {
+            "_active": true
+        },
+        "Metric-TokenMetric.XXX": {
+            "_active": true
+        },
+        "Parallelism-IncorrectCriticalRegion": {
+            "_active": true
+        },
+        "Parallelism-UnsafeVarAccess": {
+            "_active": true
+        },
+        "ProjectWide": {
+            "_active": true
+        },
+        "ProjectWideExternalAnnotations": {
+            "_active": true,
+            "project_comments": [
+                "AXIVION EXCEPTION STYLE AutosarC++19_03-A16.0.1 #line : This is a build artefact."
+            ]
+        },
+        "SaveRFG": {
+            "_active": true
+        },
+        "StaticSemanticAnalysis": {
+            "_active": true,
+            "callgraph_options": {
+                "compare_parameter_types": true,
+                "treat_uncalled_functions_as_alive": true,
+                "use_full_pointer_analysis": true
+            },
+            "enable_filtering_heuristics": true,
+            "exception_options": {
+                "missing_specifier_means_throw_any": true
+            },
+            "global_classic_options": {
+                "abstract_interpretation": true,
+                "use_pointer_filter": true,
+                "use_type_filter": true
+            },
+            "logging_level": "NONE"
+        },
+        "Stylechecks": {
+            "exclude_messages_in_system_headers": true,
+            "languages": [
+                "C",
+                "C++",
+                "C++/CLI",
+                "Assembler"
+            ]
+        }
+    },
+    "Project": {
+        "AxivionLinker": {
+            "_active": true,
+            "input_files": [
+                "$(ICEORYX_DIR)/build/install/prefix/bin/iox-cpp-publisher",
+                "$(ICEORYX_DIR)/build/install/prefix/bin/iox-cpp-subscriber",
+                "$(ICEORYX_DIR)/build/install/prefix/bin/iox-cpp-waitset-individual",
+                "$(ICEORYX_DIR)/build/install/prefix/bin/iox-roudi"
+            ],
```

Comment:
Why these selection of example here?

Reply:
The examples are the entry point for the Axivion Linker to scan the iceoryx libraries. I put here the iox-roudi for all the roudi-related libraries and examples for pub/sub and waitset for the posh lib. Axivion will then later also cover the unused functionality of the libraries.